bftengine/src/bftengine/KeyExchangeManager.cpp

```diff
@@ -259,8 +262,14 @@ void KeyExchangeManager::generateConsensusKeyAndSendInternalClientMsg(const SeqN
     candidate.pub = pub;
     candidate.cid = cid;
     candidate.sn = sn;
+    seq_candidate_map_[sn] = candidate_private_keys_;
```

Comment:
you should not generate a new candidate also when you already have the sn in the candidate's map.
(here https://github.com/vmware/concord-bft/blob/23ed4ae7f84571a12801adf0595f2b48d16e38ba/bftengine/src/bftengine/KeyExchangeManager.cpp#L251). 
In that case, we simply want to use the already-generated candidate

Reply:
made the change.